### PR TITLE
Feature/create edit remarks

### PR DESCRIPTION
**Pull request by scott062**
Updating create naming now to match FE or vice versa

https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2659

### CODE REVIEW

File: talentmap_api/fsbid/services/common.py [hunk 1]
```diff
@@ -323,6 +323,18 @@ def send_get_request(uri, query, query_mapping_function, jwt_token, mapping_func
...
+    url = f"{api_root}/{uri}"
+    response = requests.put(url, data=mappedQuery, headers={'JWTAuthorization': jwt_token, 'Content-Type': 'application/json'}).json()
+    if response.get("Data") is None or ((response.get('return_code') and response.get('return_code', -1) == -1) or (response.get('ReturnCode') and response.get('ReturnCode', -1) == -1)):
+        logger.error(f"Fsbid call to '{url}' failed.")
+        return None
+    if mapping_function:
+        return list(map(mapping_function, response.get("Data", {})))
+    else:
+        return response.get("Data", {})
+
```
Comment (scott062) [inline talentmap_api/fsbid/services/common.py:337]: Not used yet - still need to verify patterns with WS before we start using. Note the url route for this service is commented out

File: talentmap_api/fsbid/services/remark.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+
+def create_remark_and_remark_insert(query={}, jwt_token=None, host=None):
+    '''
+    Create remark and inserts
+    '''
+    hru_id = jwt.decode(jwt_token, verify=False).get('sub')
+    query['rmrkcreateid'] = hru_id
+    query['rmrkupdateid'] = hru_id
+    query['rmrkactiveind'] = 'Y'
```
Comment (scott062) [inline talentmap_api/fsbid/services/remark.py:25]: Assumption: Do not create new inactive remarks

File: talentmap_api/fsbid/services/remark.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    remark = create_remark(query, jwt_token)
+    rmrk_seq_num = pydash.get(remark, '[0].rmrk_seq_num')
+
+    if rmrk_seq_num:
+        for x in query['rmrkInsertionList']:
+            formattedInsert = {}
+            formattedInsert['ritext'] = x
+            formattedInsert['ricreateid'] = hru_id
+            formattedInsert['riupdateid'] = hru_id
+            formattedInsert['rirolerestrictedind'] = 'N'
```
Comment (scott062) [inline talentmap_api/fsbid/services/remark.py:35]: Unsure what this is except most RI are listed as 'N' - no known business rule from dev side

File: talentmap_api/fsbid/services/remark.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        "query": query,
+        "query_mapping_function": convert_remark_insert_query,
+        "jwt_token": jwt_token,
+        "mapping_function": "",
+    }
+
+    return services.send_put_request(
+        **args
+    )
+
```
Comment (scott062) [inline talentmap_api/fsbid/services/remark.py:162]: Placeholder for edit - but need to rework strategy with WS to confirm how to navigate nested resources with partial updates

File: talentmap_api/fsbid/views/remark.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    'riupdatedate': openapi.Schema(type=openapi.TYPE_STRING, description='Remark Insert Updated Date'), }
+            ), description='Remark Insertion'),
+        }
+    ))
+
+    def put(self, request):
+        '''
+        Edit remark by rmrk_seq_num
+        '''
+        return Response(services.edit_remark_and_remark_insert(request.data, request.META['HTTP_JWT']))
```
Comment (scott062) [inline talentmap_api/fsbid/views/remark.py:53]: Placeholder for edit - see above comment

**Comment by elizabeth-jimenez** [comment]
No action needed for this right now, just a reminder: delete admin_panel.py or at least the remark url in it after confirming it uses nothing after transition to fsbid/remark

**Review by elizabeth-jimenez — APPROVED** [review]

**Pull request merged**